osu.Game.Rulesets.Osu/Objects/Slider.cs

```diff
@@ -20,24 +20,33 @@ public class Slider : OsuHitObject, IHasCurve
         /// </summary>
         private const float base_scoring_distance = 100;
 
-        public IHasCurve CurveObject { get; set; }
-
-        public SliderCurve Curve => CurveObject.Curve;
+        public SliderCurve Curve { get; } = new SliderCurve();

```

Comment:
the only reason it's this way is because IHasCurve has a curve property (and it can only be a property). HOWEVER it might not be necessary now that CurvedHitObject doesn't exist anymore, let me look into it.